cypress/e2e/sauce_demo/login.cy.js

```diff
@@ -0,0 +1,124 @@
+describe('Test cases for Login page', () => {
+
+    beforeEach("load fixture", function () {
+        
+        cy.fixture("users").then((dataUsers)=>{
+            this.dataUsers=dataUsers;
+        })
+
+        cy.visit('/')
+        cy.hash().should('be.empty')
+    });
+    
+
+    it('test a correct user and a bad password', function () {
+        
+        cy.url().should('eq', 'https://www.saucedemo.com/')
+        
+        cy.get('#user-name').type(this.dataUsers.name_std)
+        cy.get('#password').type(this.dataUsers.badPass)
+        
+        cy.get('#login-button').click()
+
+        cy.get('#login_button_container > div > form > div.error-message-container.error > h3').should('contain','Epic sadface: Username and password do not match any user in this service')
```

Comment:
Looking at the page, the error element has a unique data test locator that can be used to uniquely identify the element and make the test resilient. 

The current locator you have could break if the page changes layout.

The following locator works best [data-test="error"]

Reply:
Same for all other tests that checks on this element

Reply:
Updated with Data Attribute Selectors suggestion. Thank you!